packages/modelviewer.dev/examples/postprocessing/index.html

```diff
@@ -0,0 +1,364 @@
+  <effect-composer id="selectiveComposer" render-mode="quality">
+    <selective-bloom-effect id="selectiveBloom" strength="5" radius="0.5" threshold="0"></selective-bloom-effect>
+  </effect-composer>
+</model-viewer>
+<script type="module">
+  const composer = document.querySelector("effect-composer#selectiveComposer");
+  const bloom = composer.querySelector("#selectiveBloom");
+
+  composer.addEventListener('updated-selection', () => {
+    composer.selection.forEach((obj) => 
+      (obj.material.emissive.r > 0 
+      || obj.material.emissive.g > 0 
+      || obj.material.emissive.b > 0) 
+        && bloom.selection.push(obj.name)
+    );
+  });
+</script>
+            </template>
+          </example-snippet>
+        </div>
+      </div>
+    </div>
+
+    <div class="sample">
+      <div id="combining-effects" class="demo"></div>
+      <div class="content">
+        <div class="wrapper">
+          <div class="heading">
+            <h2 class="demo-title">Combining Effects</h2>
+            <p>&lt;model-viewer-effects&gt; will attempt to join as many of your effects as possible into a single
+            <code>EffectPass</code>, which has a significant performance benefit as opposed to keeping all the effects
+            in separates Passes. Effects can be disabled by setting the <code>blend-mode</code> to <code>skip</code>.
+            <br>
+            It is recommended to follow the official <a href="https://github.com/pmndrs/postprocessing/wiki/Effect-Merging#effect-execution-order">Postprocessing Wiki</a>
+            for best practices in combining effects. The &lt;model-viewer-effects&gt; handles effect compatibility issues by 
+            placing incompatible effects in a separate Pass.
+            </p>
+          </div>
+          <example-snippet stamp-to="combining-effects" highlight-as="html">
+            <template>
+<model-viewer id="effectsViewer" camera-controls touch-action="pan-y" src="../../shared-assets/models/EmissiveStrengthTest.glb" ar alt="A 3D ">
+  <effect-composer id="composer" render-mode="quality">
+    <outline-effect color="blue" blend-mode="skip"></outline-effect>
+    <ssao-effect blend-mode="skip"></ssao-effect>
+    <glitch-effect blend-mode="skip"></glitch-effect>
+    <smaa-effect blend-mode="skip"></smaa-effect>
+    <bloom-effect blend-mode="skip"></bloom-effect>
+    <pixelate-effect granularity="5" blend-mode="skip"></pixelate-effect>
+  </effect-composer>
+  <div class="controls">
+    <label for="outline">Outline Effect</label>
+    <input type="checkbox" id="outline">
+    <label for="ssao">SSAO Effect</label>
+    <input type="checkbox" id="ssao">
+    <label for="glitch">Glitch Effect</label>
+    <input type="checkbox" id="glitch">
+    <label for="smaa">SMAA Effect</label>
+    <input type="checkbox" id="smaa">
+    <label for="bloom">Bloom Effect</label>
```

Comment:
Bloom is excellent, but do you know how to remove the hard edge at large radius?
<img width="702" alt="image" src="https://user-images.githubusercontent.com/1649964/229640512-05d09790-012d-452e-8748-de0d27e7fe17.png">

Reply:
![image](https://user-images.githubusercontent.com/54665621/229704366-abb72878-295c-4882-aaa9-cac62f15df6f.png)

I'm not seeing this artifact. I have noticed that such a hard edge occurs when combining color-grade and bloom (logically, increasing contrast or changing brightness of the canvas itself will cause such artifacts), and also banding is an issue on dark backgrounds (for which there is the option on the effect composer: render-mode="quality" which is active in this example)

Reply:
This one is consistent for me on Firefox, Chrome, and Safari (again on my M1 Macbook Pro).

Reply:
Seems to be a WebGL implementation difference then, I've been mainly testing on windows

Reply:
Writing this mainly for myself:

Safari and Chrome are embedded webkit while firefox is built on top of gecko, my assumption is that the WebGL2 implementation with Metal is different in gecko vs webkit and thats what is causing the issues. 

Test https://webglreport.com/?v=2 on the different platforms, see if any implementation is missing
Test different frameBufferTypes
Test outputEncoding sRGB vs Linear

Most likely there is not much that we can do, especially since these artifacts seem to be platform specific (IOS/windows do not have these issues). If this is too big of an issue, we can provide a flag to fail on the Metal backend using the renderer context:

```javascript
const gl = WebGLRenderer.getContext();
const debugInfo = gl.getExtension("WEBGL_debug_renderer_info");
const vendor = gl.getParameter(debugInfo.UNMASKED_VENDOR_WEBGL);     // I.e, "Google. Inc"
const renderer = gl.getParameter(debugInfo.UNMASKED_RENDERER_WEBGL); // I.e, "ANGLE (Intel, Intel(R) HD Graphics Direct3D11 vs_5_0 ps_5_0)" 
```

Additionally, I am considering adding the same WebGLRenderer flag `failIfMajorPerformanceCaveat` to the `<effect-composer>`, since model-viewer always sets it to false

Reply:
No, Chrome is only WebKit on iOS - on Mac it's Blink, just like Chrome on Windows & Linux. I feel your pain - I've debugged many driver-specific GPU rendering issues. But this is the main value that model-viewer provides: a consistent cross-platform rendering experience, so our users don't have to worry about things like this. There must be some users of `postprocessing` who have run into this - hopefully there are already some workarounds described. If not, we need to file a bug on them. 

I'm also curious if this is a general Mac problem, or specific to the M1 chip.

Reply:
Getting this as well on my macbook air (both chrome and firefox). I'll try to debug and see if theres a difference threejs/postprocessing wise with regards to buffers, its most likely an alpha blend issue.
Interestingly I'm only getting this with the bloom-effect, the selective-bloom-effect works fine on mac on both firefox and chrome without that hard edge.

Reply:
@mrdoob This might be an interesting one to bring up with your GPU Chromies. I know color profiles are still fairly recent, so I imagine there are some wrinkles to iron out still.

Reply:
![image](https://user-images.githubusercontent.com/54665621/231318901-c0e84228-f4d1-41ad-b741-f6748534a5de.png)

I have reproduced it on windows as well by setting the Color Profile to anything but the default **sRGB IEC61966-2.1**, so this isn't a platform specific issue at all.

Reply:
Good to know... have you filed a bug on `postprocessing` yet?

Reply:
https://github.com/pmndrs/postprocessing/issues/475

Reply:
Okay, interesting discussion on that bug. Anyway, your rocket looks perfect for me now. This example we've been talking about still has the issue, unless I select either of the Reinhard2 tonemappers, in which case it looks great. Especially considering this is restricted to a relatively small number of devices and there are various workarounds and fixes probably coming down the pipe, I think this is looking just fine. Thanks for the effort!